dts/arc/synopsys/emsk.dtsi

```diff
@@ -20,12 +20,30 @@
 			compatible = "snps,arcem";
 			reg = <0>;
 		};
+	};
 
-		intc: arcv2-intc {
-			compatible = "snps,arcv2-intc";
-			interrupt-controller;
-			#interrupt-cells = <2>;
-		};
+	intc: arcv2-intc {
+		compatible = "snps,arcv2-intc";
+		interrupt-controller;
+		#interrupt-cells = <2>;
+	};
+
+	timer0: timer0 {
+		compatible = "snps,arc-timer";
+		interrupts = <16 1>;
+		interrupt-parent = <&intc>;
+	};
+
+	timer1: timer1 {
+		compatible = "snps,arc-timer";
+		interrupts = <17 1>;
+		interrupt-parent = <&intc>;
+	};
+
+	sectimer0: sectimer0 {
+		compatible = "snps,arc-timer";
+		interrupts = <20 1>;
+		interrupt-parent = <&intc>;
```

Comment:
Hmm that looks not entirely correct - this interrupt exists not on all boards, see (from soc.h)

```
#if defined(CONFIG_BOARD_EM_STARTERKIT_R23) && defined(CONFIG_SOC_EMSK_EM7D)
#define IRQ_SEC_TIMER0				20
#endif /* CONFIG_BOARD_EM_STARTERKIT_R23 && CONFIG_SOC_EMSK_EM7D */
```

but dts/arc/synopsys/emsk.dtsi is used for all boards.

And on some configurations this irq line is for `IRQ_CORE_DMA_COMPLETE` interrupt.

Reply:
`IRQ_SEC_TIMER0` seems to be used only in `drivers/timer/arcv2_timer0.c`, only if `CONFIG_ARC_SECURE_FIRMWARE=y`, so from what I understand, certain boards can't enable this option. This needs to be specified somehow (probably not by adding board dependencies, but an auxiliary Kconfig option like `HAS_ARC_SEC_TIMER` or whatever else, selected at SoC level).

I can still define this node at board level, making it 100% equivalent to the previous code, but to me, this shows that there's still an issue somewhere.

Reply:
see latest change: [`ff72f58` (#46585)](https://github.com/zephyrproject-rtos/zephyr/pull/46585/commits/ff72f58de4a019784c6629f5173dfd64f5a4ad2c)